### PR TITLE
PR: Make sure processEvents is not called by Pdb

### LINKED ISSUE #10834: Keyboard events leak from console to editor while debugging
### What steps will reproduce the problem?

create a file `tmp.py`:
```
def a():
    1/0
```
And a file `tmp2.py`:
```
from tmp import a
a()
```
 1. Run tmp2.py
 2. type in the console `%debug`
 3. Type `u+<Return>` and close `tmp.py`
 4. Type `d` without return
 5. Press return as many times and as quickly as possible.
 
I see 1-2 new lines being added to `tmp2.py` before `tmp.py` has time to load and the focus returns to the console.

## Versions

* Spyder version: 4.0.0.dev0 e5f2892d9
* Python version: 3.7.5
* Qt version: 5.12.6
* PyQt5 version: 5.12.3
* Operating System: Darwin 19.0.0

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.4.2 (OK)
qtconsole >=4.6.0            :  4.7.0.dev0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.5 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.1 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.0;<2.0.0:  1.9.0.dev0 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.9.0 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.4 (OK)
scipy >=0.17.0               :  1.3.3 (OK)
pyls >=0.31.0;<0.32.0        :  0.31.1 (OK)
applaunchservices >=0.1.7    :  0.2.1 (OK)
```

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Calling QApplication.processEvents() creates a leak of keypresses. This Pr makes sure this doesn't happen again.

### Issue(s) Resolved

Fixes #10834
Depends on spyder-ide/spyder-kernels#185

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by ccordoba12** [comment]
@impact27, do you think this is critical for Spyder 4? Or can we leave for 4.1?

**Comment by impact27** [comment]
It can be quite annoying while debugging, especially when stepping through multiple files, as one of the files might end up with an extra 'n' or something.

File: continuous_integration/azure/install.bat [hunk 1]
```diff
@@ -38,7 +38,7 @@ conda create -n jedi-test-env -q -y python=3.6 loghub spyder-kernels -c spyder-i
 if errorlevel 1 exit 1
 
 :: Install spyder-kernels from master
-pip install -q --no-deps git+https://github.com/spyder-ide/spyder-kernels
+pip install -q --no-deps git+https://github.com/spyder-ide/spyder-kernels.git@refs/pull/185/head
```
Comment (ccordoba12) [inline continuous_integration/azure/install.bat]: I just merged spyder-ide/spyder-kernels#185, so you can revert these changes.
